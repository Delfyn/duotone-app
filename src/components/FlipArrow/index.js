/* @flow */

import React from 'react';
import './style.css';

type Props = {
  className?: string,
  size?: number,
  onClick?: Function,
};

const FlipArrow = ({ className = '', size = 40, onClick }: Props) => (
  <svg
    className={className}
    onClick={onClick}
    width={size}
    height={size}
    viewBox={`0 0 ${size} ${size}`}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    <defs>
      <circle id="a" cx={size / 2} cy={size / 2} r={size / 2} />
      <mask id="b" x="0" y="0" width={size} height={size} fill="#fff">
        <use xlinkHref="#a" />
      </mask>
    </defs>
    <g fill="none" fillRule="evenodd">
      <use stroke="#979797" mask="url(#b)" strokeWidth="4" xlinkHref="#a" />
      <g fill="#9B9B9B">
        <path d="M8 20l5-7v14zM13 21v-2h14v2zM32 20l-5 7V13z" />
      </g>
    </g>
  </svg>
);

export default FlipArrow;
