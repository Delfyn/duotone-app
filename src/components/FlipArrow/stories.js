/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import FlipArrow from './index';

storiesOf('FlipArrow', module)
  .add('Simple example', () => (
    <FlipArrow onClick={action('onClick')} />
  ));
