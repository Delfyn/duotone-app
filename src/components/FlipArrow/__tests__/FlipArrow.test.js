import React from 'react';
import renderer from 'react-test-renderer';
import FlipArrow from '../index';

test('FlipArrow renders correctly', () => {
  const component = renderer.create(<FlipArrow />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
