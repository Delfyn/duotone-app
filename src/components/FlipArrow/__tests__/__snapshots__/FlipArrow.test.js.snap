// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlipArrow renders correctly 1`] = `
<svg
  className=""
  height={40}
  onClick={undefined}
  viewBox="0 0 40 40"
  width={40}
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <circle
      cx={20}
      cy={20}
      id="a"
      r={20}
    />
    <mask
      fill="#fff"
      height={40}
      id="b"
      width={40}
      x="0"
      y="0"
    >
      <use
        xlinkHref="#a"
      />
    </mask>
  </defs>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <use
      mask="url(#b)"
      stroke="#979797"
      strokeWidth="4"
      xlinkHref="#a"
    />
    <g
      fill="#9B9B9B"
    >
      <path
        d="M8 20l5-7v14zM13 21v-2h14v2zM32 20l-5 7V13z"
      />
    </g>
  </g>
</svg>
`;
